homeassistant/components/mobile_app/http_api.py

```diff
@@ -0,0 +1,80 @@
+"""Provides an HTTP API for mobile_app."""
+from typing import Dict
+
+from aiohttp.web import Response, Request
+
+from homeassistant.auth.util import generate_secret
+from homeassistant.components.cloud import (async_create_cloudhook,
+                                            async_is_logged_in)
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.components.http.data_validator import RequestDataValidator
+from homeassistant.const import (HTTP_CREATED, HTTP_INTERNAL_SERVER_ERROR,
+                                 CONF_WEBHOOK_ID)
+
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.storage import Store
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (ATTR_REGISTRATIONS, ATTR_SUPPORTS_ENCRYPTION,
+                    CONF_CLOUDHOOK_ID, CONF_CLOUDHOOK_URL, CONF_SECRET,
+                    CONF_USER_ID, DOMAIN, REGISTER_DEVICE_SCHEMA)
+
+from .helpers import supports_encryption, savable_state
+
+from .webhook import register_device_webhook
+
+
+def register_http_handlers(hass: HomeAssistantType, store: Store) -> bool:
+    """Register the HTTP handlers/views."""
+    hass.http.register_view(DevicesView(store))
+    return True
+
+
+class DevicesView(HomeAssistantView):
+    """A view that accepts device registration requests."""
+
+    url = '/api/mobile_app/devices'
+    name = 'api:mobile_app:register-device'
+
+    def __init__(self, store: Store) -> None:
+        """Initialize the view."""
+        self._store = store
+
+    @RequestDataValidator(REGISTER_DEVICE_SCHEMA)
+    async def post(self, request: Request, data: Dict) -> Response:
+        """Handle the POST request for device registration."""
+        hass = request.app['hass']
+
+        resp = {}
+
+        webhook_id = generate_secret()
+
+        if async_is_logged_in(hass):
+            cloudhook = await async_create_cloudhook(hass, webhook_id)
+
+            if cloudhook is not None:
+                data[CONF_CLOUDHOOK_ID] = resp[CONF_CLOUDHOOK_ID] = \
```

Comment:
For this code, instead of maintaining `resp` while you're going through the code, consider defining resp at the end:

```python
{
  CONF_CLOUDHOOK_ID: data[CONF_CLOUDHOOK_ID],
  # etc
}
```

Reply:
Also make a copy of `data`.